### ISSUE TITLE
Retire the JAX LP-IPM (lp_ipm.py) — second half of the #364 fallback-chain retirement

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/discopt#370]
## Summary

Retire the pure-JAX LP interior-point method (`discopt/_jax/lp_ipm.py`, `lp_ipm_solve` / `lp_ipm_solve_batch`) — the second half of #364's "retire the fallback chain." The first half (the JAX LP-IPM last resort in the standalone `_solve_lp`) landed in #368. Verification (below) shows the remaining JAX-IPM uses are all either (a) fallbacks for a **no-Rust-extension install** that discopt does not actually support, or (b) forward solves already covered by POUNCE / the Rust simplex.

**POUNCE stays** — it is the NLP solver for MINLP subproblems *and* the interior-point forward solver the differentiable layers require (see below). This issue removes only the JAX IPM.

## Why JAX-IPM is not needed

**Differentiability is already IFT / implicit-KKT, not unrolled-IPM.** `_jax/differentiable_lp.py` computes LP gradients via `@jax.custom_jvp` (OptNet-style implicit differentiation of the KKT conditions) with the **forward solve on POUNCE** (`solve_lp_kkt`, via `jax.pure_callback`). QP (`differentiable_qp.py`) and NLP/MINLP are likewise implicit-KKT / envelope-theorem. No gradient path unrolls the IPM.

Note the differentiable layers need an **interior / analytic-center** forward point (a simplex *vertex* has a degenerate active set → singular KKT sensitivity). POUNCE provides it; the Rust simplex cannot be used there. Since POUNCE stays, this is already handled.

## Verification: `lp_ipm_solve_batch` is a fallback, not the default path

In `_solve_milp_bb` (the spatial/MINLP B&B), node-relaxation solving is gated:

```python
node_engine = "simplex"                       # default
_simplex_nodes = bool(SIMPLEX_AVAILABLE)       # True in any built install
_structured_avail = _simplex_nodes or (prefer_pounce and _pounce_nodes_avail)
if _structured_avail:      # Path B — Rust exact-vertex warm simplex per node  ← DEFAULT
    _node_solve(...)       # = _solve_node_lp_simplex
elif n_batch > 1:          # JAX-IPM batch  ← only when the Rust simplex is UNavailable
    lp_ipm_solve_batch(...)
```

`SIMPLEX_AVAILABLE` is True whenever the compiled `discopt._rust` core is present — i.e. always, in a built wheel. So Path B (the Rust warm-started simplex hardened in #364) is the default per-node engine, and `lp_ipm_solve_batch` is only reachable in a no-Rust install, which is not a supported configuration.

## Reroute map (what to change)

| JAX-IPM site | current role | route to |
|---|---|---|
| `solver.py::_solve_milp_bb` `lp_ipm_solve_batch` (~11016) | node-LP fallback | drop the `elif`; Path B (Rust simplex) is default. If a no-structured-engine guard is still wanted, error clearly. |
| `solver.py::_root_cover_cut_loop` `lp_ipm_solve` (~10281) | root LP then "cross to vertex" | **Rust simplex** (returns the vertex directly, no crossover) |
| `_jax/differentiable_solve.py` forward/`_solve_objective` `lp_ipm_solve` (108, 183, 229) | forward solve in the unified dispatcher | **POUNCE** (`solve_lp_kkt` / `lp_pounce`), matching `differentiable_lp.py` — appears redundant with it already |
| `_jax/differentiable_solve.py` `qp_ipm_solve` (if same pattern) | QP forward | POUNCE QP (out of scope if QP-IPM is separate; note it) |

After the consumers are rerouted, **delete `_jax/lp_ipm.py`** (and `lp_ipm_solve_batch`), and drop the now-dead no-Rust batch branch.

## What we are NOT removing

- **POUNCE** anywhere — NLP engine + differentiable-layer forward solver. Kept.
- The QP IPM (`qp_ipm.py`) unless it is confirmed to be in the same redundant position — audit separately.
- `differentiable_lp.py` / `differentiable_qp.py` — the implicit-KKT gradient machinery is the keeper.

## Throughput footnote (not a blocker)

`lp_ipm_solve_batch` could GPU-parallelize many node LPs in one vmap call, whereas Path B loops per node (sequential Rust simplex). But Path B is already the chosen default (warm-started, exact-vertex, KKT-valid bound), and a genuine Rust *batch* path exists (`solve_lp_batch_py` / `crates/discopt-core/src/lp/simplex/batch.rs`) if batched throughput is ever wanted — so no niche requires the JAX IPM. Adopting the Rust batch solver in Path B is a separate perf item.

## Validation plan

- Reroute each site; run the full soundness gauntlet + the BARON comparison sweep (must stay 0 violations).
- Confirm the differentiable-optimization consumers (`estimate.py`, `_jax/parametric.py`, `doe/*`, `pounce_layer.py`) still produce correct gradients via the POUNCE + custom_jvp path.
- Grep-confirm no remaining `lp_ipm` import before deleting `lp_ipm.py`.

## References
- #364 (retire the fallback chain), #368 (first half: JAX LP-IPM removed from `_solve_lp`).
- Differentiable layers: `_jax/differentiable_lp.py` (POUNCE + custom_jvp), `_jax/differentiable_qp.py`.
- Default node engine gate: `solver.py` `_structured_avail` / `_simplex_nodes` (~10731, ~10940, ~10960).
